client/engine/engine.go

```diff
@@ -372,7 +372,11 @@ func (e *Engine) handleObjectiveRequest(or protocols.ObjectiveRequest) (EngineEv
 		return e.attemptProgress(&vfo)
 
 	case virtualdefund.ObjectiveRequest:
-		vdfo, err := virtualdefund.NewObjective(request, true, myAddress, e.store.GetChannelById, e.store.GetConsensusChannel)
+		bal, err := e.vm.Balance(request.ChannelId)
+		if err != nil {
```

Comment:
Won't we get an error if we call this as Irene?

Reply:
I guess not since your `client_test` covers this. I think the reason is that Irene _does_ register the channel with her voucher manager. Is that what we want to happen? It seems a bit strange for intermediaries to do that -- they might not even want to have a voucher manager at all, I would have thought.

Reply:
I think you're correct that this doesn't fail because Irene does register the channel as well.

>It seems a bit strange for intermediaries to do that -- they might not even want to have a voucher manager at all, I would have thought.

I think you're right, I'll update so Irene doesn't store vouchers.

Reply:
I've updated this in [1f5f8a2](https://github.com/statechannels/go-nitro/pull/866/commits/1f5f8a2a7575d17290bd8da6d1c3d3f775d43614) so Irene no longer stores vouchers or attempts to call `e.vm.Balance`